tests/unit/moduleapi/cross-version-cluster.tcl

```diff
@@ -33,24 +42,18 @@ tags {external:skip needs:other-server cluster modules singledb} {
 
             test "Send cluster message via module from latest to other server" {
                 assert_equal OK [r test.pingall]
-                assert_match "*cluster_stats_messages_module_sent:1*" [r CLUSTER INFO]
-                wait_for_condition 50 100 {
-                    [string match {*cluster_stats_messages_module_received:1*} [r -1 CLUSTER INFO]]
-                } else {
-                    fail "Node didn't receive the message"
-                }
+                wait_for_cluster_stat 0 messages_module_sent 1
+                wait_for_cluster_stat 1 messages_module_received 1
+                wait_for_cluster_stat 1 messages_module_sent 1
+                wait_for_cluster_stat 0 messages_module_received 1
             }
 
-            test "Send cluster message via module from other server to latest" {
-                r CONFIG resetstat
-                r -1 CONFIG resetstat
+            test "Send cluster message via module from other to latest server" {
                 assert_equal OK [r -1 test.pingall]
-                assert_match "*cluster_stats_messages_module_sent:1*" [r -1 CLUSTER INFO]
-                wait_for_condition 50 100 {
-                    [string match {*cluster_stats_messages_module_received:1*} [r CLUSTER INFO]]
-                } else {
-                    fail "Node didn't receive the message"
-                }
+                wait_for_cluster_stat 1 messages_module_sent 2
+                wait_for_cluster_stat 0 messages_module_received 2
+                wait_for_cluster_stat 0 messages_module_sent 2
+                wait_for_cluster_stat 1 messages_module_received 2
```

Comment:
I was going to look into this further, I observed it while updating the cross version test to work with other versions and latest as well. https://github.com/valkey-io/valkey/pull/2342. Thanks for deep diving on this.

Reply:
I think however we should have 2 followup items:
1. make config resetstats reset the cluster message stats
2. fix the test module so it will always provide null terminated node-id.
Will try to remember open an issue/PR tomorrow